packages/@blockprotocol/graph/src/stdlib/subgraph/edge/link-entity.ts

```diff
@@ -0,0 +1,225 @@
+import { Entity, EntityId } from "../../../types/entity.js";
+import { OutwardEdge, Subgraph } from "../../../types/subgraph.js";
+import {
+  isHasLeftEntityEdge,
+  isHasRightEntityEdge,
+  isIncomingLinkEdge,
+  isOutgoingLinkEdge,
+  OutgoingLinkEdge,
+} from "../../../types/subgraph/edges/outward-edge-alias.js";
+import { mustBeDefined } from "../../must-be-defined.js";
+import { getEntity } from "../element/entity.js";
+
+const convertTimeToStringWithDefault = (
+  timestamp?: Date | string,
+  defaultConstructor = () => new Date().toISOString(),
+) => {
+  return timestamp === undefined
+    ? defaultConstructor()
+    : typeof timestamp === "string"
+    ? timestamp
```

Comment:
I'm not sure we should do this, it could cause strange behaviour with timezones and other parsing things. I think we should just require in documentation that the string is ISO formatted